### PR TITLE
ASoC: Intel: sof_sdw: tag SoundWire BEs as non-atomic

### PR DESCRIPTION
**Pull request by plbossart**
The SoundWire BEs make use of 'stream' functions for .prepare and
.trigger. These functions will in turn force a Bank Switch, which
implies a wait operation.

Mark SoundWire BEs as nonatomic for consistency, but keep all other
types of BEs as is.

Suggested-by: Takashi Iwai <tiwai@suse.de>
Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

### CODE REVIEW

**Review by ranj063 — DISMISSED** [review]

**Review by bardliao — APPROVED** [review]

File: sound/soc/intel/boards/sof_sdw.c [hunk 1]
```diff
@@ -588,14 +588,15 @@ static void init_dai_link(struct device *dev, struct snd_soc_dai_link *dai_links
 			  struct snd_soc_dai_link_component *cpus, int cpus_num,
 			  struct snd_soc_dai_link_component *codecs, int codecs_num,
 			  int (*init)(struct snd_soc_pcm_runtime *rtd),
-			  const struct snd_soc_ops *ops)
+			  const struct snd_soc_ops *ops, bool nonatomic)
```
Comment (RanderWang) [inline sound/soc/intel/boards/sof_sdw.c]: one more args for init_dai_link. Can we make it "static inline void", then args count is not a problem.

File: sound/soc/intel/boards/sof_sdw.c [hunk 2]
```diff
@@ -935,7 +936,7 @@ static int create_sdw_dailink(struct device *dev, int *be_index,
 			      playback, capture,
 			      cpus + *cpu_id, cpu_dai_num,
 			      codecs, codec_num,
-			      NULL, &sdw_ops);
+			      NULL, &sdw_ops, true);
```
Comment (kv2019i) [inline sound/soc/intel/boards/sof_sdw.c]: Adding booleans to init_dai_link() makes it even harder to follow. How about just setting nonatomic to true here in a separate call outside init_dai_link (with a comment that notes SDW uses nonatomic stream ops and thus the nonatomic flag needs to be set)?  No need to modify other places where init_dai_link is called.

**Review by kv2019i — CHANGES_REQUESTED** [review]
Hmm, init_dai_link() is starting to be ready hard to follow.

File: sound/soc/intel/boards/sof_sdw.c (hunk 2 above)
Comment (plbossart) [reply sound/soc/intel/boards/sof_sdw.c]: good suggestion @kv2019i, I just made that change.

**Review by RanderWang — APPROVED** [review]

**Review by ranj063 — APPROVED** [review]

**Review by bardliao — APPROVED** [review]
